src/main/java/com/google/sps/servlets/UserSettingsServlet.java

```diff
@@ -0,0 +1,82 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.JsonObject;
+import com.google.sps.data.User;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+// So far this servlet allows fetching and uploading a display name.
+@WebServlet("/settings")
+public class UserSettingsServlet extends HttpServlet {
+  // This regex checks that user names are alphanumeric and (5 - 25) characters.
+  static final String namePattern = "^[\\w_ -]{5,25}";
```

Comment:
nit: can you use all caps for constant names?
https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names